### PR TITLE
Adds support for checking out revisions on top of tags and branches

### PR DESCRIPTION
**Pull request by lweijl**
@Bladrak I am not much of a Ruby programmer, so if the code needs improvement please state this out and I will do my best in fixing that!

Thanks.

### CODE REVIEW

File: lib/capistrano/rsync.rb [hunk 1]
```diff
@@ -8,9 +8,17 @@
...
 
-# Sparse checkout allows to checkout only part of the repository
+# Merely here for backward compatibility reasons
 set_if_empty :rsync_checkout_tag, false
 
+# Option states what to checkout
+set_if_empty :rsync_checkout, "branch"
+
+# To be somewhat backward compatible to the previous checkout_tag option
+if fetch(:rsync_checkout_tag, false)
```
Comment (Bladrak) [inline lib/capistrano/rsync.rb:18]: Well done :ok_hand:

File: lib/capistrano/rsync.rb [hunk 2]
```diff
@@ -40,12 +48,24 @@
 end
 
 rsync_target = lambda do
-  target = !!fetch(:rsync_checkout_tag, false) ? "tags/#{fetch(:branch)}" : "origin/#{fetch(:branch)}"
+  if fetch(:rsync_checkout) == "tag"
```
Comment (Bladrak) [inline lib/capistrano/rsync.rb]: Maybe you could use a `case` statement here, as follows:

``` ruby
case fetch(:rsync_checkout)
when "tag"
    ...
when "revision"
    ...
else
    ...
```

**Comment by Bladrak** [comment]
@lweijl could you also add an entry in the changelog file? Thanks! Seems great so far! :)

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,6 @@
+## 1.3.1 (Jul 15, 2015)
```
Comment (Bladrak) [inline CHANGELOG.md]: Actually, this would be for 1.3.3 (sorry :/) ; Could you also specify that rsync_checkout_tag is now deprecated? Thanks again!

**Comment by Bladrak** [comment]
@lweijl feel free to notify me once you edited your PR, I'm not notified by github otherwise :)

**Comment by lweijl** [comment]
@Bladrak Hopefully this triggers that notification than ;-)

**Comment by Bladrak** [comment]
@lweijl it did indeed, thanks a lot!

**Pull request merged**